### DISCUSSION TITLE
Updating an enum not working

### DISCUSSION THREAD

**Discussion by KnowYourselves** [prisma/prisma#8070] [Q&A] [answered]
Hello! I have the following code:
![image](https://user-images.githubusercontent.com/26102039/124443656-dde52600-dd4b-11eb-99e2-cb290b772ab6.png)
My problem is that the status isn't changing, even after the update it stays the same.

I have no idea why this isn't working as I've done the same on other parts of the same project and it has worked. Any help would be appreciated.

**Comment by ryands17** [comment, marked as answer]
@KnowYourselves 👋 

Could you share your schema so that I can reproduce?

**Reply by KnowYourselves** [reply]
Hello!

```js
enum PurchaseOrderStatus {
  CREATED
  ACCEPTED
  REJECTED
  CANCELLED
  FULFILLED
  PROCESSING
  WAITING
  DEFUNCT
}

model PurchaseOrder {
  id                 String              @id
  deliveryDate       DateTime
  quantity           Int
  quantityDispatched Int                 @default(0)
  unitPrice          Float
  channel            String
  status             PurchaseOrderStatus @default(CREATED)
  notes              String?
  rejection          String?
  canceled           String?
  urlNotification    String?
  createdAt          DateTime
  updatedAt          DateTime
}
```

**Reply by ryands17** [reply]
Works fine for me on Postgres and the latest version of Prisma. Could you update `prisma` and `@prisma/client` to the latest version and check? If that still doesn't work, I would suggest opening an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so that we can look into this.

![image](https://user-images.githubusercontent.com/19697099/124480392-d3b33e00-ddc4-11eb-8bb4-d33c3b7498c3.png)

**Reply by KnowYourselves** [reply]
Thanks!

After digging a ton I realized this only happens in production. The DEFUNCT Status is undefined in the status object in production.  

I used `prisma migrate deploy` and the migration does in did exists and was applied. Is there a way I could correct this without resetting the database?

Thx!

**Reply by ryands17** [reply]
Did you run `prisma generate` after `prisma migrate deploy`? It's possible that the change you made wasn't generated.

**Reply by KnowYourselves** [reply]
And that did indeed solve it (I thought it ran automatically) thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
